perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -110,6 +103,20 @@ static void PrintFlags(const bool* FlagArray)
     std::cout << OutputBuffer << '\n';
 }
 
+static size_t ComputeEffectiveAddress(const instruction_operand& Operand, const s16* Registers)
```

Comment:
Not super important, but I would label this function as inline. It doesn't matter too much for the compiler, but it matters for the readers of the code: you say that this is a simple one-liner that should be thought of as going inline in call sites, not some complex function that will do crazy things. It can be useful to know this bit of information before looking at the function body. And sometimes the compiler may also listen.

I write this taking into account that you might refactor it, following my other comments. If you leave it like this, then it's definitely alright to not use inline.

Reply:
1cf78e683ce906ab23d5f276eba2560eddd97632
I left it as 3 lines to avoid a potential subexpression overflow, so I have 2 questions
1) Does it lessen the chances the compiler will actually inline the function?
2) If it does, is it worth it to cast the operands to avoid overflows?

Reply:
1. First, think about what the CPU is doing. `Result = a + b + c;` is two additions whether it's one expression or 3 expressions. So ultimately, no difference for the CPU. Now as for the compiler? I honestly don't know. Usually compilers represent statements as an abstract syntax tree (AST), and the AST might look different if it chooses to represent operators `+=` and `+` differently. If the AST is different, then it could make different decisions. But I'm not knowledgeable at all about this. At any rate, you don't really need to know this, because it's trivial to check: just look at the asm and see if there's a call instruction or not.
2. No, don't bother. If you really want it inlined, instead of playing mind games with your compiler, most compilers give an explicit way to force inlining. For MSVC it would be `__forceinline`. The downside is that it's specific to one compiler.

Reply:
Also note that there is no reason to avoid overflow. It's stored as 32 bits in your program. But the data is not 32 bits, it's 8 or 16 bits. That's all it can ever be, because it's the only thing an 8086 can use. So if you add as 32 bits and then truncate to 16 bits at some point losing the highest bits, or cast to 16 bits before adding and it overflows, it's the same.

Reply:
Ah right, I figured since it's an effective address in a Megabyte array an overflow would be a problem.
I forgot the way to access the whole memory in an actual 8086 is with the segment registers.
That's the piece I was missing and why I thought overflow mattered. Thanks!